### PR TITLE
Handling financial year

### PR DESCRIPTION
**Pull request by alfretoine**
## Self proof reading checklist

- [ ] Did I run Pronto and fixed all warnings
- [ ] Make sure all naming and code will remain understandable in 6 month by someone new to the project or by you.
- [ ] Did I test the right thing?
- [ ] Did I test corner cases, non happy path (defensive testing)?
- [ ] Check that I used the ad-hoc patterns and created files accordingly
- [ ] Check code efficiency with record intensive project
- [ ] Update documentation,readme accordingly

Thanks!

### CODE REVIEW

**Comment by mestachs** [comment]
add a more "system spec" with a package with one activity and verify depending on quarter that the correct dhis2 values

File: spec/lib/orbf/rules_engine/services/period_iterator_spec.rb [hunk 1]
```diff
@@ -22,5 +22,36 @@
...
+      expect(described_class.periods("2018Q1", "financial_july")).to eq(
+        ["2017July"]
+      )
+    end
+
+    it "converts month to financial_july" do
+      expect(described_class.periods("201803", "financial_july")).to eq(
+        ["2017July"]
+      )
+
```
Comment (mestachs) [inline spec/lib/orbf/rules_engine/services/period_iterator_spec.rb]: what about 201807 ?

https://en.wikipedia.org/wiki/Off-by-one_error
```
     def first_date
          anniv_date = range.first.beginning_of_year + 6.months
         range.first < anniv_date ? (anniv_date - 1.year) : anniv_date
      end
```
or
```
     def first_date
          anniv_date = range.first.beginning_of_year + 6.months
         range.first <= anniv_date ? (anniv_date - 1.year) : anniv_date
      end
```
